### ISSUE TITLE
Incorrect types - schema with z.date() does not result in FieldGetter

### ISSUE THREAD

**Issue by mcky** [esamattis/react-zorm#14]
Hi,

I've got a field with a date picker that I'm casting from a string to a date with `z.preprocess`, but TS incorrectly warns me that I can't call the field name getter
```
<input type="date" name={form.fields.date()} />
                                     ^^^^^^
This expression is not callable.
  Type 'FieldChain<{ toString: {} [..OMITTED..] }>' has no call signatures.
```

`form.errors.date` seems to work fine however

[Codesandbox repro](https://codesandbox.io/s/confident-glitter-tudziz?file=%2Fsrc%2FApp.tsx)

Happy to attempt a PR patching it if you'd like and can point me in the right direction

**Comment by esamattis** [comment]
Thanks for the report! It is now fixed and released in v0.3.2

[v0.3.2](https://github.com/esamattis/react-zorm/blob/master/packages/react-zorm/CHANGELOG.md#v032)

**Comment by mcky** [comment]
Amazing, thanks! Loving the library by the way
